Add autocompletion for attribute values

TimvdLippe opened the pull request:
Add support for autocompletion for attributes values added in the editor service. You can use `nested.html` to test it out. `ctrl + space` for an attribute. Select either `inherit-please` or `non-notifying-property`. Based on these, the autocompletions for the value (hit `ctrl + click` again) are `{{foo}},{{bar}}` and `[[foo]],[[bar]]` respectively.

Requires https://github.com/Polymer/polymer-editor-service/pull/38

 - [x] CHANGELOG.md has been updated

TimvdLippe commented:
As the upstream PR has been merged, this PR is not blocked anymore :smile:

rictic commented:
Hm, I can't reproduce the feature as described. I open `nested.html`, and when I autocomplete inside of the behavior-test-elem's attributes section I do get autocompletions for its attribute names, but when I select e.g. `inherit-please` it autocompletes to `inherit-please="number"`. And when I delete `number` and hit ctrl+space (either with or without {{}}s) I don't get any autocompletions.

We need tests, badly. :/

TimvdLippe commented:
Hm that's really weird, it still works for me. I will have to write tests, but that is blocked on #53

TimvdLippe commented:
I have now added fancy tests to confirm this is working as intended :tada: 

The file was formatted with CRLF, so I had to fix that to LF. This makes the diff hard to read :( Most importantly, the last two tests are added and I extracted 2 constants to make positioning a bit easier if you are modifying the fixture.

One thing I discovered is that it autocompletes to all properties. We could optimize this to filter on the property type. E.g. if the local property is of type String, only autocomplete the using element properties which are also Strings.

TimvdLippe commented on spec/auto-completer-spec.js:
```diff
@@ -1,150 +1,205 @@
+
+          waitsForPromise(async () => {
+            var result = await getCompletions();
+            expect(result.length).toBe(2);
```
I actually took them from #54 and maintained the same structure. Fine by me to transform them all into `deepEqual` 😄
